File: teamolympiad.java
import java.util.Scanner;
import java.util.ArrayList;

public class teamolympiad{

	public static void main(String[]args){
	
		Scanner in = new Scanner(System.in);

		int n = in.nextInt();
		ArrayList <Integer> prog = new ArrayList<Integer>();
		ArrayList <Integer> math = new ArrayList<Integer>();
		ArrayList <Integer> pe = new ArrayList<Integer>();

		for(int i = 0; i < n; i++){
			int t = in.nextInt();

			if(t == 1){
				prog.add(i+1);
			}
			
			if(t == 2){
				math.add(i+1);
			}
			
			if(t == 3){
				pe.add(i+1);
			}
		}

		int teams = Math.min(Math.min(prog.size(), math.size()), pe.size());
		System.out.println(teams);

		for(int i = 0; i < teams; i++){
			System.out.println(prog.get(i)+ " " + math.get(i)+ " " + pe.get(i));

		}
	}
}

File: teamolympiad.py
n = int(input())
kids = list(map(int, input().split()))
prog = []
math = []
pe = []

for i in range(n):
    if kids[i] == 1:
        prog.append(i+1)
    if kids[i] == 2:
        math.append(i+1)
    if kids[i] == 3:
        pe.append(i+1)

teams = min(min(len(prog), len(math)), len(pe))
print(teams)

for i in range(teams):
    print(f"{prog[i]} {math[i]} {pe[i]}")




